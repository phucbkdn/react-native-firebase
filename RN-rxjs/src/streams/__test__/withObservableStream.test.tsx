import React from 'react'
import withObservableStream from '../index'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'
import { BehaviorSubject } from 'rxjs'
import { map } from 'rxjs/operators'

describe('Testing withObservableStream', () => {
  it('Should render correctly', (done) => {
    const Component = ({ count }) => <Text>{count}</Text>

    const observable = new BehaviorSubject({ count: 1 })
    const HOCComponent = withObservableStream(
      observable,
      {},
      { count: 1 }
    )(Component)
    const view = renderer.create(<HOCComponent />)
    expect(view).toMatchSnapshot()
    const text = view.root.findByType(Text)
    expect(text.props.children).toBe(1)
    done()
  })
})
