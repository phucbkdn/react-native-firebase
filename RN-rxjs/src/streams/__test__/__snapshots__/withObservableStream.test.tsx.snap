// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing withObservableStream Should render correctly 1`] = `
<Text>
  1
</Text>
`;
